keras_cv/models/densenet.py

```diff
@@ -0,0 +1,304 @@
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      weights: one of `None` (random initialization), or a pretrained
+        weight file path.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      pooling: optional pooling mode for feature extraction
+        when `include_top` is `False`.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional block.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional block, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+      name: (Optional) name to pass to the model.  Defaults to "DenseNet".
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if not (weights in {None} or tf.io.gfile.exists(weights)):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` or the path to the weights file to be loaded."
+        )
+
+    if include_top and None in input_shape:
+        raise ValueError(
+            "If `include_top` is True, "
+            "you should specify a static `input_shape`. "
+            f"Received: input_shape={input_shape}"
+        )
+
+    if include_top and not classes:
+        raise ValueError(
+            "If `include_top` is True, "
+            "you should specify `classes`. "
+            f"Received: classes={classes}"
+        )
+
+    inputs = layers.Input(shape=input_shape)
+
+    x = inputs
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
+
+    x = layers.ZeroPadding2D(padding=((3, 3), (3, 3)))(x)
+    x = layers.Conv2D(64, 7, strides=2, use_bias=False, name="conv1/conv")(x)
+    x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="conv1/bn")(x)
+    x = layers.Activation("relu", name="conv1/relu")(x)
+    x = layers.ZeroPadding2D(padding=((1, 1), (1, 1)))(x)
+    x = layers.MaxPooling2D(3, strides=2, name="pool1")(x)
+
+    x = DenseBlock(x, blocks[0], name="conv2")
+    x = TransitionBlock(x, 0.5, name="pool2")
+    x = DenseBlock(x, blocks[1], name="conv3")
+    x = TransitionBlock(x, 0.5, name="pool3")
+    x = DenseBlock(x, blocks[2], name="conv4")
+    x = TransitionBlock(x, 0.5, name="pool4")
+    x = DenseBlock(x, blocks[3], name="conv5")
+
+    x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="bn")(x)
+    x = layers.Activation("relu", name="relu")(x)
+
+    if include_top:
```

Comment:
IMO you may want to inline the two functions below, since they are only used once and there existence make the model less copy/pastable (you have to remember to copy them too)